readseeker.go

```diff
@@ -0,0 +1,153 @@
+package desync
+
+import (
+	"fmt"
+	"io"
+	"sort"
+)
+
+// TODO: Implement WriterTo interface
+//       default io.Copy implementation operates in 32k chunks; copying up to a full chunk at a time will improve perf
+
+// IndexPos represents a position inside an index file, to permit a seeking reader
+type IndexPos struct {
+	Store  Store
+	Index  Index
+	Length int64 // total length of file
+	pos    int64 // Location within offset stream; must be 0 <= Pos <= Index.
+
+	curChunkID     ChunkID // hash of current chunk
+	curChunk       []byte  // decompressed version of current chunk
+	curChunkIdx    int     // identity of current chunk
+	curChunkOffset int64   // offset within current chunk
+}
+
+func NewIndexReadSeeker(i Index, s Store) IndexPos {
```

Comment:
Given the nature of this reader, may want to consider returning *IndexPos instead